master/internal/api_project.go

```diff
@@ -214,7 +214,32 @@ func (a *apiServer) getProjectColumnsByID(
 	}
 
 	summaryMetricsSet := make(map[string]struct{})
+	validationMetricsSet := make(map[string]struct{})
 	for _, trial := range summaryMetrics {
+		if rawMetrics, ok := trial.SummaryMetrics["validation_metrics"]; ok {
+			metrics := rawMetrics.(map[string]interface{})
+			metricsKeys := make([]string, 0, len(metrics))
+			for metricKey := range metrics {
+				metricsKeys = append(metricsKeys, metricKey)
+			}
+			sort.Strings(metricsKeys)
+
+			for _, key := range metricsKeys {
+				if _, seen := validationMetricsSet[key]; !seen {
```

Comment:
if a trial has "x" reporting string metrics and another trial has "x" reporting numeric metrics we will return the type with a smaller ID. 

Is this ok? I'm not quite sure how this API is used.

Reply:
This is expected. We are assuming there should only be relevant/comparable experiments within the same project for the table to make more sense.

Reply:
I wonder if some people are still using ```Uncategorized``` for everything and this makes that assumption not always true. I just would be concerned if we have potential for type cast errors or anything like that down the line.

Reply:
I get what you are saying, even though this information might not be useful, we still shouldn't provide incorrect information. I changed it so that if we have multiple metrics with the same name in the same group, the type becomes unspecified.